backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/migration/MigrationTask.java

```diff
@@ -90,17 +98,20 @@ public Instant timestamp() {
     private final CassandraSchemaVersionDAO schemaVersionDAO;
     private final CassandraSchemaTransitions transitions;
     private final SchemaVersion target;
+    private final List<String> achievedMigrations;
 
     @VisibleForTesting
-    public MigrationTask(CassandraSchemaVersionDAO schemaVersionDAO, CassandraSchemaTransitions transitions, SchemaVersion target) {
+    public MigrationTask(CassandraSchemaVersionDAO schemaVersionDAO, CassandraSchemaTransitions transitions, SchemaVersion target, List<String> achievedMigrations) {
         this.schemaVersionDAO = schemaVersionDAO;
         this.transitions = transitions;
         this.target = target;
+        this.achievedMigrations = achievedMigrations;
```

Comment:
create a new mutable list  which is a copy of the parameter. Maybe a concurrent one ?

Reply:
Why do you think it would be better?